File: motion_planning/scripts/MotionPlanningPublisher.py
#!/usr/bin/env python

import rospy
from nav_msgs.msg import Path

class MotionPlanningPublisher:
    """
    Publishes trajectory messages describing the robot's desired trajectory.

    The robot's trajectory, composed of an array of positions, orientations, and times, is represented by nav_msgs/Path messages published to the motion_planning/trajectory topic.

    Attributes:
        _pathPub: Publisher to motion_planning/trajectory. Publishes nav_msgs/Path messages.
    """

    def __init__(self):
        """Initialize MotionPlanningPublisher instance."""
        self._pathPub = rospy.Publisher('motion_planning/trajectory', Path, queue_size=10)

    def publishTrajectory(self, trajectory):
        """
        Publish a nav_msgs/Path message.

        Uses the _pathPub Publisher to publish to motion_planning/trajectory.

        Args:
            trajectory: The nav_msgs/Path message.
        """
        self._pathPub.publish(trajectory)

File: motion_planning/src/MotionPlanningPublisher.cc
#include "motion_planning/MotionPlanningPublisher.hh"

namespace MuddSub::MotionPlanning
{
  MotionPlanningPublisher::MotionPlanningPublisher(ros::NodeHandle n):
    n_(n)
  {
    motionPlanningPub_ = n_.advertise<nav_msgs::Path>("motion_planning/trajectory", 1000);
  }

  void MotionPlanningPublisher::publishTrajectory(nav_msgs::Path& trajectory)
  {
    motionPlanningPub_.publish(trajectory);
  }
}
